updated datepicker documentation

xman343 opened the pull request:
<!--
Thank you for your contribution to iTwinUI.

If you are only making changes to the docs site using the "Edit page on GitHub" link,
then you can ignore most of this template.
-->

## Changes

<!--
What kind of code changes does this PR include?
Mention anything that could be helpful for reviewers and include screenshots for visual changes.
-->

Edited docs pages for the DatePicker component to bring it up-to-date with current implementation. Created new examples based on storybook.

## Testing

<!--
How did you test your changes?
If your PR has visual changes, then make sure they are demonstrated in html test pages as well as
storybook, then approve visual test images for both (`yarn approve:css` and `yarn approve:react`).

If not applicable, you can write "N/A".
-->

N/A

## Docs

<!--
If your PR includes user-facing changes, then update docs in all places (JSDoc, website, stories, etc).
Make sure to include a changeset (`yarn changeset`).

If not applicable, you can write "N/A".
-->

N/A

gretanausedaite commented on apps/website/src/pages/docs/datepicker.mdx:
```diff
@@ -22,36 +22,64 @@ import * as AllExamples from '~/examples';
+### Basic
 
-The default date picker is a default menu button from which an interactive calendar expands on click. This allows the user to visually select the date rather than entering the date with a keyboard. This has the advantage of also showing weekdays, which may be important in a given context. The date picker is commonly used within column filters.
+The basic date picker consists of a default menu button from which an interactive calendar expands on click.
```
The example does not have menu button from which a calendar expands. Maybe just describe that basic DatePicker consists of month skipping and date picking.
Then you could add a "Usage" section at the bottom of docs page with information that users can add DatePickers to menu buttons or simply on the page.

xman343 replied:
Edited the description & added usage section.

gretanausedaite replied:
Add code example how to use DatePicker with menu button.

xman343 replied:
Added!

gretanausedaite commented on apps/website/src/pages/docs/datepicker.mdx:
```diff
@@ -22,36 +22,64 @@ import * as AllExamples from '~/examples';
+### Date + time selection
 
-There are two ways to select a date with the text field date picker; clicking on the button and selecting a date, or typing it out using the format shown in the help message underneath the field (DD-MM-YYYY). The date automatically fills the text field when directly selected in the calendar and is editable. It’s a more convenient way to offer editing in a context where the user might need to chan ...
+If you need to include the time, there are multiple ways to do so. Below is the default variant, which uses twenty-four hour / military time.
```
Remove "military-time", I think 24-hour format is enough.

xman343 replied:
Fixed.

gretanausedaite commented on apps/website/src/pages/docs/datepicker.mdx:
```diff
@@ -22,36 +22,64 @@ import * as AllExamples from '~/examples';
+The basic date picker consists of a default menu button from which an interactive calendar expands on click.
 
-### Text field
+<LiveExample src='DatePicker.basic.tsx' truncate={false}>
```
`truncate={false}` means that the code takes full space it needs, we usually do that only on main example. Other code examples should be truncated.

xman343 replied:
Fixed.

gretanausedaite reviewed (commented):
You can remove `<Placeholder />` component. It's only temporary to tell people that docs are not finished.

Also, could you keep 1 component update per 1 PR? It's easier to track what's changed.

gretanausedaite commented:
Could you merge this PR into `main` branch? We could release documentation update faster than major release.

xman343 commented:
> You can remove `<Placeholder />` component. It's only temporary to tell people that docs are not finished.
> 
> Also, could you keep 1 component update per 1 PR? It's easier to track what's changed.

Removed the placeholder component and reverted the commit for the unrelated component

adamhock commented on apps/website/src/pages/docs/datepicker.mdx:
```diff
@@ -14,44 +14,74 @@ import * as AllExamples from '~/examples';
 
 <p>{frontmatter.description}</p>
 
-<Placeholder componentPageName='core-datepicker' />
```
Since this `Placeholder` was removed, the `Placeholder` import statement can also be removed

xman343 replied:
removed

adamhock commented on apps/website/src/pages/docs/datepicker.mdx:
```diff
@@ -14,44 +14,74 @@ import * as AllExamples from '~/examples';
+</LiveExample>
 
-There are two ways to select a date with the text field date picker; clicking on the button and selecting a date, or typing it out using the format shown in the help message underneath the field (DD-MM-YYYY). The date automatically fills the text field when directly selected in the calendar and is editable. It’s a more convenient way to offer editing in a context where the user might need to chan ...
+This allows the user to visually select the date rather than entering the date with a keyboard. This has the advantage of also showing weekdays, which may be important in a given context. The date picker is commonly used within column filters.
```
```suggestion
This allows the user to visually select the date rather than entering the date with a keyboard. This has the advantage of also showing weekdays, which may be important in a given context. The date picker is commonly used within column filters in Table.
```
Also make sure to link `Table` to the Table documentation. There's an example of how to do this in the Expandable Block documentation page

xman343 replied:
done

adamhock replied:
Make sure it links to the Table documentation like so

```mdx
[Table](table)
```

xman343 replied:
Fixed!

adamhock commented on apps/website/src/pages/docs/datepicker.mdx:
```diff
@@ -14,44 +14,74 @@ import * as AllExamples from '~/examples';
+
 ## Props
 
 <PropsTable path={frontmatter.propsPath} />
```
There should be a Related Component section (see Expandable Block)

xman343 replied:
added

adamhock commented on apps/website/src/pages/docs/datepicker.mdx:
```diff
@@ -14,44 +14,74 @@ import * as AllExamples from '~/examples';
+
+## Usage
+
+A DatePicker can be featured on your page in multiple ways. One can be added directly to your page so that it is visible as soon as the page loads. Additionally, a DatePicker can be added to a menu button on your page so that the user can toggle its visibility.
```
```suggestion
A date picker can be featured on your page in multiple ways. One can be added directly to your page so that it is visible as soon as the page loads. Additionally, a date picker can be added to a menu button on your page so that the user can toggle its visibility.
```

xman343 replied:
done

adamhock reviewed (commented):
I would change the title of the section from `DatePicker` to `Date picker` (see Color picker)

These comments are just from reading the documentation. 

I was getting some build errors when building this branch, but they could be from my end. I will follow up on this tomorrow

xman343 commented:
> I would change the title of the section from `DatePicker` to `Date picker` (see Color picker)
> 
> These comments are just from reading the documentation.
> 
> I was getting some build errors when building this branch, but they could be from my end. I will follow up on this tomorrow

fixed!

mayank99 commented on apps/website/src/examples/index.tsx:
```diff
@@ -242,9 +249,6 @@ export { default as ToggleSwitchInputGroupExample } from './ToggleSwitch.inputgr
 export { default as TooltipMainExample } from './Tooltip.main';
 export { default as TooltipPlacementExample } from './Tooltip.placement';
 
-export { default as TransferListMainExample } from './TransferList.main';
-export { default as TransferListWithLabelExample } from './TransferList.withlabel';
-
```
revert this change. build is breaking because of it.

xman343 commented:
> 
I agree, will keep `dev` as the target
